clippy_lints/src/redundant_test_prefix.rs

```diff
@@ -0,0 +1,74 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use clippy_utils::{is_in_cfg_test, is_in_test_function};
+use rustc_hir::intravisit::FnKind;
+use rustc_hir::{Body, FnDecl};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::declare_lint_pass;
+use rustc_span::def_id::LocalDefId;
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Triggers when a function annotated with the `#[test]` macro begins with `test_`
+    /// ### Why is this bad?
+    /// This is redundant, since the annotations already denotes the function a test.
+    /// Tests are executed using `cargo test `module::test-name-here::`.
+    /// It's clear that a test is a test, without prefixing `test_` to the test name.
+    /// ### Example
+    /// ```no_run
+    /// #[test]
+    /// fn my_test_case() {
```

Comment:
```suggestion
    /// fn test_my_test_case() {
```
I assume you're meaning to show a 'bad' example here that should trigger the lint.

Reply:
Ah, right, I need to update this.